packages/@coorpacademy-components/src/template/skill-detail/all-courses.js

```diff
@@ -0,0 +1,159 @@
+import ButtonLink from '../../atom/button-link';
+import SearchForm from '../../molecule/search-form';
+import CardsGrid from '../../organism/cards-grid';
+import searchValueIncluded from '../../util/search-value-included';
+import InputSwitch from '../../atom/input-switch';
+import style from './all-courses.css';
+
+const FilterButton = (props, context) => {
+  const {selected, filter, onClick} = props;
+  const {skin} = context;
+  const primarySkinColor = get('common.primary', skin);
+
+  const buttonProps = {
+    customStyle: {
+      backgroundColor: selected ? primarySkinColor : '#E1E1E3',
+      color: selected ? '#FFFFFF' : '#515161',
+      transition: 'background-color 0.15s ease-in-out, color 0.15s ease-in-out',
+      width: 'fit-content'
+    },
+    label: filter,
+    onClick,
+    'data-name': 'filter-type-course-button'
+  };
+
+  return <ButtonLink {...buttonProps} />;
+};
+
+FilterButton.contextTypes = {
+  skin: Provider.childContextTypes.skin,
+  translate: Provider.childContextTypes.translate
+};
+
+FilterButton.propTypes = {
+  selected: PropTypes.bool,
+  filter: PropTypes.string,
+  onClick: PropTypes.func
+};
+
+const AllCourses = (props, context) => {
+  const {skillIncludedCourses, filters, sorting} = props;
+  const {options, onChange} = filters;
+  const {list} = skillIncludedCourses;
+  const {translate} = context;
+  const [showCompleted, setShowCompleted] = useState(true);
+  const [searchValue, setSearchValue] = useState('');
+  const [searchResults, setSearchResults] = useState(
+    list.sort((a, b) => {
+      const progressA = a.progress ?? 0;
+      const progressB = b.progress ?? 0;
+      return progressB - progressA;
+    })
+  );
+
+  const sortView =
+    sorting !== undefined ? (
+      <div data-name="choice">
+        <Select {...sorting} aria-label="All courses sort" />
+      </div>
+    ) : null;
+
+  const handleSearch = useCallback(
+    value => {
+      setSearchValue(value);
+      setSearchResults(list.filter(skill => searchValueIncluded(skill.title, value)));
+    },
+    [list, setSearchValue, setSearchResults]
+  );
+
+  const handleSearchReset = useCallback(() => {
+    setSearchValue('');
+    setSearchResults(list);
+  }, [list, setSearchValue, setSearchResults]);
+
+  const handleShowCompletedToggle = useCallback(() => {
+    const tempShowCompleted = !showCompleted;
+    setShowCompleted(tempShowCompleted);
+    if (tempShowCompleted) {
+      setSearchResults(list);
+      handleSearchReset();
+    } else {
+      setSearchResults(searchResults.filter(skill => skill.progress === 0));
+    }
+  }, [searchResults, showCompleted, setShowCompleted]);
+
+  return (
+    <>
+      <div className={style.continueLearningWrapper}>
+        <span className={style.continueLearningTitle}>{translate('all_courses')}</span>
+        <span className={style.continueLearningNumber}>{skillIncludedCourses.list.length}</span>
+      </div>
+      <div className={style.searchAndSortSection}>
```

Comment:
:bulb: @wernerdurand Wouldn't a completed courses be relevant? 

as **activity page** is filterable by playlist, not skills. 
And this is more than relevant that users see their existing progress. Maybe big card would be too much, but this is something we might discuss

Reply:
I've asked and waiting for Werner's feedback on this and the one above

Reply:
👌  (we can keep this idea for later)

Reply:
This is a very good idea, but I think we can skip it for now and add it later!